src/Extractor/ReadAccessor.php

```diff
@@ -193,4 +251,36 @@ public function getExtractCallback(string $className): ?Expr
             new Arg(new Scalar\String_($className)),
         ]);
     }
+
+    /**
+     * Get AST expression for binding closure when dealing with a private property.
+     */
+    public function getExtractIsNullCallback(string $className): ?Expr
+    {
+        if ($this->type !== self::TYPE_PROPERTY || !$this->private) {
```

Comment:
maybe we can extract a private method like `needsExtractIsNullCallback()` used here and in `getIsNullExpression()`? this would make explicit that these cases are linked

Reply:
On the other method it's just a sub case, i think it makes it more abstract for little gain will keep it like this